PyRoute/Galaxy.py

```diff
@@ -327,11 +334,18 @@ def set_area_alg(self, star, area, algs):
             area.alg.setdefault(star.alg_code, Allegiance(full_alg.code, full_alg.name, base=False)).worlds.append(star)
 
     def set_positions(self):
+        shadow_len = self.stars.number_of_nodes()
         for sector in self.sectors.values():
             for star in sector.worlds:
-                self.stars.add_node(star)
+                if star not in self.ranges:
```

Comment:
Again, have you encountered this condition?

Reply:
See earlier reply